### PR TITLE
timeseries streaming from eventhub into datalake 

### PR DESCRIPTION
**Pull request by kristianschneider**
Create and deploy databricks workspace and streaming cluster
Create streaming job to consume events from eventhub
#373

### CODE REVIEW

File: source/databricks/package/eventhub_ingestor.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    spark = SparkSession.builder.getOrCreate()
+
+    input_configuration = {}
+    input_configuration["eventhubs.connectionString"] = spark.sparkContext._gateway.jvm.org.apache.spark.eventhubs.EventHubsUtils.encrypt(event_hub_connection_key)
+    streamingDF = (spark.readStream.format("eventhubs").options(**input_configuration).load())
+
+    checkpoint_path = f"abfss://{delta_lake_container_name}@{storage_account_name}.dfs.core.windows.net/checkpoint"
+
+    streamingDF.writeStream.option("checkpointLocation", checkpoint_path).foreachBatch(lambda df: process_eventhub_item(df, events_delta_path)).start()
+    # streamingDF.awaitTermination()
```
Comment (ZavezX) [inline source/databricks/package/eventhub_ingestor.py:44]: Remove?

File: build/databricks_streaming_cluster/main/main.tf [hunk 2]
```diff
@@ -42,20 +42,18 @@ resource "databricks_job" "streaming_job" {
...
 
   spark_python_task {
     python_file = "dbfs:/timeseries/streaming.py"
     parameters  = [
-   # params for job will go here
+         "--data-storage-account-name=${data.azurerm_key_vault_secret.st_data_lake_name.value}",
+         "--data-storage-account-key=${data.azurerm_key_vault_secret.st_data_lake_primary_access_key.value}",
+         "--event-hub-connection-key=${data.azurerm_key_vault_secret.evh_timeseries_listen_connection_string.value}",
+         "--delta-lake-container-name=${data.azurerm_key_vault_secret.st_data_lake_data_container_name.value}",
+         "--events-data-blob-name=timeseries"
```
[the remark this answers is not in the document]
Comment (lasrinnil) [reply build/databricks_streaming_cluster/main/main.tf:56]: That sounds like a good idea, could you perhaps post a link or some other reference for how the naming should be?

[the remark this answers is not in the document]
Comment (ZavezX) [reply build/databricks_streaming_cluster/main/main.tf:56]: I don't necessary mean that we should use bronze, silver or gold in the naming. I just think that the name timeseries is not informative enough to reflect which stage/state the data is in. But the medallion architecture is shown [here](https://databricks.com/notebooks/delta-lake-cdf.html#:~:text=The%20medallion%20architecture%20takes%20raw,many%20customers%20implement%20in%20Databricks.).

File: source/databricks/streaming-jobs/streaming.py [hunk 3]
```diff
@@ -10,4 +10,28 @@
 # distributed under the License is distributed on an "AS IS" BASIS,
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
-# limitations under the License.
\ No newline at end of file
+# limitations under the License.
+
```
Comment (lasrinnil) [inline source/databricks/streaming-jobs/streaming.py:14]: Could this file have a more describing name? 

streaming.py -> timeseries_persistor.py

File: build/databricks_streaming_cluster/main/kv.tf [hunk 4]
```diff
@@ -19,4 +19,19 @@ data "azurerm_key_vault" "kv_shared" {
...
+
+data "azurerm_key_vault_secret" "st_data_lake_primary_access_key" {
+  name         = "st-data-lake-primary-access-key"
+  key_vault_id = data.azurerm_key_vault.kv_shared.id
+}
+
+data "azurerm_key_vault_secret" "evh_timeseries_listen_connection_string" {
+  name         = "evh-timeseries-listen-connection-string"
+  key_vault_id = data.azurerm_key_vault.kv_shared.id
+}
```
Comment (ZavezX) [inline build/databricks_streaming_cluster/main/kv.tf:37]: Shouldn't this be saved in the keyvault in timeseries and not in shared?

**Review by ZavezX — CHANGES_REQUESTED** [review]

**Review by ZavezX — APPROVED** [review]

**Pull request merged**